sis-local/README.md

```diff
@@ -0,0 +1,72 @@
+# sis-local
+
+Emulate SiS functionality in a local environment. This may not work perfectly, but it should catch many common errors
+and provide a much smoother experience doing local development and porting to SiS!
+
+- Throw warnings for commands that don't work in SiS
+- (optional) Handles local Snowpark session setup
+
+**If you find some missing compatibility issue, please file an issue and tag `@sfc-gh-jcarroll`**
+
+## Usage
+
+### Installation
+
+Just like Snowpark, **it requires python 3.8.**
+
+```shell
+pip install "streamlit_in_snowflake @ git+https://github.com/sfc-gh-jcarroll/sis-tricks.git@sis-local#subdirectory=sis-local"
+```
+
+In general, we recommend managing your environment with Conda,
+as it will transition easily to the SiS environment. However, this package is not yet available in the [Snowflake Anaconda channel](https://repo.anaconda.com/pkgs/snowflake/),
+so it may be useful to install separately once your conda environment is setup.
+
+See the example directory for a minimal conda setup. Read more in the
+[SiS docs](https://docs.snowflake.com/en/LIMITEDACCESS/streamlit-in-snowflake#installing-packages-manually).
+
+### Magic code block for the top of your app
+
+```python
```

Comment:
This would be a useful use-case of the walrus operator, but that's probably not worth using since
1. It's not very familiar
2. get_active_session() is probably fast to return

But, just for fun, I tried rewriting it, and it seemed reasonable. Mostly I just wanted an excuse to try out how the walrus operator feels, because I've rarely ever used it :) 

```python
from __future__ import annotations

from snowflake.snowpark.context import get_active_session
from snowflake.snowpark import Session
from snowflake.snowpark.exceptions import SnowparkSessionException

def get_sis_session() -> Session | None:
    try:
        return get_active_session()
    except SnowparkSessionException:
        return None

if session := get_sis_session():
    import streamlit as st
else:
    import streamlit_in_snowflake as st
    # This part is optional, and requires secrets.toml setup
    from streamlit_in_snowflake.local_session import get_local_session
    session = get_local_session()
```

Reply:
Neat! I wouldn't have thought to use this either.

For the sake of tribal knowledge - I _believe_ the way that get_active_session() is written today, it's quite efficient to call multiple times since it constructs the session prior to calls (somehow) https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/session.py#L144-L151